src/Board/Board.ts

```diff
@@ -876,6 +876,86 @@ module jsflap.Board {
             };
         }
 
+        public toJSON(): string {
+            //Contributed by Jason Ma
+            //Creates a string in JSON format of the graph and returns it
+            //Note apparently this function is not the one called when you try to export graphs
+            //It's another function in the jsflap.js file.
+            var JSONData = '';
+
+            var bounds = this.getBounds();
+
+            var minX = bounds.minX,
+                maxX = bounds.maxX,
+                minY = bounds.minY,
+                maxY = bounds.maxY;
+
+            var offsetPoint = new Point.IMPoint(minX, minY);
+
+            JSONData += 'Graph {\n';
+
+            JSONData += '\tNodes {\n';
+            this.visualizations.nodes.forEach((node: Visualization.NodeVisualization) => {
+                var pos = node.position.getMPoint().subtract(offsetPoint).round();
+                JSONData += '\t\t' + node.model.label + '{\n';
+                JSONData += '\t\t\tXCoor : ' + pos.x + '\n';
+                JSONData += '\t\t\tYCoor : ' + pos.y + '\n';
+                if(node.model.initial) {
+                    JSONData += '\t\t\tInitial : True\n'';
+                } else {
+                    JSONData += '\t\t\tInitial : False\n';
+                }
+                if(node.model.final) {
+                    JSONData += '\t\t\tFinal : True\n';
+                } else {
+                    JSONData += '\t\t\tFinal : False\n';
+                }
+                JSONData += '\t\t}'
+            });
+            JSONData += '\t}';
+
+            JSONData += '\tEdges {';
+
+            JSONData += '\t}';
+            JSONData += '}';
+            return JSONData;
+
+        }
+
+        public toJFLAP(): string {
+            //Contributed by Jason
```

Comment:
You may want to look here for programatic XML in JS: https://developer.mozilla.org/en-US/docs/Web/Guide/Parsing_and_serializing_XML